lib/plugins/bull-mq-metrics/queueDiscoverers.ts

```diff
@@ -0,0 +1,38 @@
+import { AbstractBackgroundJobProcessor } from '@lokalise/background-jobs-common'
+import type { Redis } from 'ioredis'
+
+export type QueueDiscoverer = {
+  discoverQueues: () => Promise<string[]>
+}
+
+export class RedisBasedQueueDiscoverer implements QueueDiscoverer {
+  constructor(
+    private readonly redis: Redis,
+    private readonly queuesPrefix: string,
+  ) {}
+
+  async discoverQueues(): Promise<string[]> {
+    const scanStream = this.redis.scanStream({
+      match: `${this.queuesPrefix}:*:meta`,
+    })
+
+    const queues = new Set<string>()
+    for await (const chunk of scanStream) {
+      // eslint-disable-next-line no-extra-semi
```

Comment:
Can't we remove this extra semi?

Reply:
Prettier wants it and eslint doesn't.

I'm not going to fix that within this PR.

Reply:
sounds good, let's revise after we switch to biome